src/components/navigation/quickbar.tsx

```diff
@@ -38,7 +37,14 @@ export function Quickbar({ subject, className, placeholder }: QuickbarProps) {
   const overlayWrapper = useRef<HTMLDivElement>(null)
 
   useEffect(() => {
-    if (query && !data) fetchData()
+    const effectFetch = async () => {
+      const fetchedData = await fetchQuickbarData(subject)
+      if (fetchedData) setData(fetchedData)
+    }
+
+    if (query && !data) {
+      effectFetch().catch(console.error)
```

Comment:
I don't like the `effectFetch` name as a function should convey what it's doing and not in which context it's invoked. I propose a name like "fetchAndSetQuickbarData" or as it's only used in one place and you are using a similar name in the closure below, just get rid of the function.

```suggestion
   if (query && !data) {
     fetchQuickbarData(subject).then((fetchedData) => fetchedData && setData(fetchedData)).catch(console.error)
   }
```

Reply:
While I agree with your reasoning about naming and unnecessary functions: wrapping async calls in a function is annoyingly the correct way of using async code inside of `useEffect` as far as I know. If you know a cleaner way without the linter screaming at me I'm very interested ;)

Reply:
Just using promises without async/await is the cleanest (edited my suggestion just now), but generally I also like async/await syntax better. I think in this example it fits well enough into a oneliner.

Reply:
While I try to avoid `.then`, I agree it's more concise here. Thanks.